### PR TITLE
PI-2172 - Implement End-to-End Test for "View Case" in Create and Vary a Licence (CVL)

### PR DESCRIPTION
**Pull request by smalepati99**
This pull request adds an end-to-end test for the "View Case" functionality in Create and Vary a Licence (CVL). The test scenario covers the following steps:

1. Given a Case Reference Number (CRN) with the creation and booking of the NOMIS case & sentence.
2. Check if the CRN is in the right state (no active licence to release, and the offender is still in prison) using an API Call.
3. Recall the prisoner to put the CRN in the right state using an API Call.
4. Discard all existing licences using an API Call.
5. Login to Delius and delete the existing licence conditions.
6. Login to CVL as a Probation Officer and create a Licence for the CRN.
7. Login to CVL as a Prison Officer and approve the Licence for the CRN.
8. Release the prisoner to apply the licence conditions using an API Call.
9. Login to Delius and navigate to Licence Condition.
10. Verify that the licence condition appears in Delius.

### CODE REVIEW

File: steps/api/cvl/cvl-api.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { type APIRequestContext, Page, request } from '@playwright/test'
```
Comment (marcus-bcl) [inline steps/api/cvl/cvl-api.ts]:
```suggestion
import { type APIRequestContext, request } from '@playwright/test'
```

File: steps/api/cvl/cvl-api.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+export const discardAllLicences = async function (crn: string){
+   const licenceIds = await getLicenceIds(crn)
+    for (const id of licenceIds) {
+        await discardLicence(id);
+    }
+}
+
+export const discardLicence = retry(
+    sanitiseError(async (licenceId: string) => {
+        const response = await (
```
Comment (marcus-bcl) [inline steps/api/cvl/cvl-api.ts]:
```suggestion
        await (
```

File: steps/cvl-licences/application.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import { expect, Page } from '@playwright/test'
+import { faker } from '@faker-js/faker'
+import {login as cvlLogin, loginAsPrisonOfficer} from "./login.js";
+
+export const createLicence = async (page: any, crn: string, nomsNumber: string) => {
```
Comment (marcus-bcl) [inline steps/cvl-licences/application.ts]:
```suggestion
export const createLicence = async (page: Page, crn: string, nomsNumber: string) => {
```

File: steps/cvl-licences/application.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    await page.getByLabel(/Who is the initial appointment with?/).fill(faker.person.fullName())
+    await page.getByRole('button', { name: 'Continue' }).click()
+    await expect(page).toHaveTitle('Create and vary a licence - Create a licence - Where is the initial appointment?')
+    await page.getByLabel(/Building and street/).first().fill(faker.location.buildingNumber() + " " + faker.location.street())
+    await page.getByLabel(/Town or city/).fill(faker.location.city())
+    await page.getByLabel('County (optional)').fill(faker.location.county());
+    await page.getByLabel(/Postcode/).fill(faker.location.zipCode())
+    await page.getByRole('button', { name: 'Continue' }).click()
+    await expect(page).toHaveTitle('Create and vary a licence - Create a licence - What is the contact phone number for the initial appointment?')
+    await page.getByLabel('UK telephone number').fill('01632 960901');
```
Comment (marcus-bcl) [inline steps/cvl-licences/application.ts]: Should we pass a phone number generated by Faker here?

File: steps/cvl-licences/application.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    await expect(page).toHaveTitle('Create and vary a licence - Create a licence - Additional Post Sentence Supervision Conditions')
+    await page.getByLabel(/No/).check()
+    await page.getByRole('button', { name: 'Continue' }).click()
+    await expect(page).toHaveTitle('Create and vary a licence - Create a licence - Check your answers')
+    await page.getByRole('button', { name: /Send to prison for approval/ }).click()
+    await expect(page).toHaveTitle('Create and vary a licence - Create a licence - Confirmation')
+    await page.getByRole('link', { name: /Sign out/ }).click()
+};
+
+export const approveLicence = async (page: any, crn: string, nomsNumber: string) => {
```
Comment (marcus-bcl) [inline steps/cvl-licences/application.ts]:
```suggestion
export const approveLicence = async (page: Page, crn: string, nomsNumber: string) => {
```

File: steps/delius/licence-condition/delete-licence-condition.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    await expect(page.locator('h1')).toHaveText('Licence Conditions');
+
+    // Get all "delete" links
+    const deleteLinks = await page.locator('#licenceConditionTable [title="Delete licence condition"]').all();
+
+    // Iterate over each delete link and click on it
+    for (const deleteLink of deleteLinks) {
+        await deleteLink.click();
+        // Add a short delay to allow the page to process the deletion
+        await page.waitForTimeout(1000);
```
Comment (marcus-bcl) [inline steps/delius/licence-condition/delete-licence-condition.ts]: Do you need this explicit wait?  Doesn't the expect on the next line implicitly wait until the header appears?

File: tests/create-and-vary-a-licence-and-delius/view-case.spec.ts [hunk 7]
```diff
@@ -1,16 +1,51 @@
...
+
+    // Verify that the licence condition appears in Delius.
+    await deliusLogin(page);
+    await navigateToLicenceConditions(page, crn);
+    await refreshUntil(page, () =>
+            expect(page.locator(deliusLicenceCondition)).toHaveText('Standard 9 Conditions - Standard 9 Conditions'),
+        {timeout: 120_000});
+});
+
+const resetLicenceCase = async (page: Page) => {
```
Comment (marcus-bcl) [inline tests/create-and-vary-a-licence-and-delius/view-case.spec.ts]:
```suggestion
const resetLicenceCase = async () => {
```

**Review by marcus-bcl — COMMENTED** [review]
Few minor formatting issues - might be worth running:
```shell
npx eslint . --fix
```
to fix them all.

Otherwise looks great!

**Review by pmcphee77 — APPROVED** [review]

**Pull request merged**